components/collection/drop/GenerativePreview.vue

```diff
@@ -112,6 +112,15 @@ const props = defineProps<{
 
 const emit = defineEmits(['generation:start', 'generation:end', 'mint'])
 const { imageDataPayload, imageDataLoaded } = useGenerativeIframeData()
+
+const { start: startTimer } = useTimeoutFn(() => {
+  // quick fix: ensure that even if the completed event is not received, the loading state of the drop can be cleared
+  if (!props.mintCountAvailable && !imageDataLoaded.value) {
+    isLoading.value = false
+    emit('generation:end')
+  }
+}, 3000)
```

Comment:
It's not enough because sometimes drop could take 10+ s.

I would suggest adding the check that this drop has been minted out, so it only apply to `old` drop.

Reply:
`!props.mintCountAvailable` check should be enough?

Reply:
Okay, could you add a comment that it is only for old drops?

Reply:
What if the user trigger the variation button again?

Reply:
@Jarsen136 done